Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,793 @@
+                } else {
+                    if ( $Version -ge 10 ) {
+                        $filterList += $serverName
+                    } else {
+                        Write-Warning "$server is not a Windows version with AMSI support."
+                    }
+                }
+            } else {
+                Write-Warning "$server is not an Exchange version with AMSI support."
+            }
+        }
+    } else {
+        foreach ($server in $ServerList) {
+            $serverName = $server.Split('.')[0]
+            if ( $fullList -contains $serverName ) {
+                if ( $SupportedExchangeServers.Name -contains $serverName ) {
+                    $Version = GetWindowsMayorVersion -ExchangeServer $server
+                    if ( $Version -eq 0 ) {
+                        Write-Warning "We could not get Windows version for $server."
+                        Write-Warning "Try to run the script locally."
+                    } else {
+                        if ( $Version -ge 10 ) {
+                            $filterList += $serverName
+                        } else {
+                            Write-Warning "$server is not a Windows version with AMSI support."
+                        }
+                    }
+                } else {
+                    Write-Warning "$server is not an Exchange version with AMSI support."
+                }
+            } else {
+                Write-Warning "We did not find any Exchange server with name: $server"
+                if ( $TestAMSI ) {
+                    $filterList += $server
+                }
+            }
+        }
+    }
+
+    if ( ( ( $filterList.count -gt 0 ) -or
+            $TestAMSI -or
+           ( ( $EnableAMSI -or $DisableAMSI) -and
+            -not $ServerList ) ) -and
+        $SupportedExchangeServers.count -gt 0 ) {
+
+        if ($TestAMSI) {
+            foreach ($server in $filterList) {
+                Write-Host $bar
+                Write-Host ""
+                Write-Host "Testing $($server):" -ForegroundColor Magenta
+                Write-Host ""
+                if ( $fullList -contains $server ) {
+                    CheckServerAMSI -ExchangeServer $server -isServer
+                } else {
+                    CheckServerAMSI -ExchangeServer $server
+                }
+                Write-Host ""
+            }
+        }
+
+        if ($CheckAMSIConfig) {
+            if ($filterList) {
+                foreach ( $server in $filterList) {
+                    Write-Host $bar
+                    Write-Host ""
+                    Write-Host "Checking $($server):" -ForegroundColor Magenta
+                    Write-Host ""
+                    CheckAMSIConfig -ExchangeServer $server
+                    Write-Host ""
+                }
+            }
+            Write-Host $bar
+            Write-Host ""
+            $getSO = $null
+            $getSO = Get-SettingOverride -ErrorAction SilentlyContinue | Where-Object {
```

Comment:
We have the `Get-FilteredSettingOverrideInformation.ps1` function that we use in `HealthChecker` to filter out `Setting Overrides`. However, the function is not yet a shared function but could be simply changed to become one.
Might be useful for this script and other future scripts as well. What do you think @dpaulson45

Reply:
Resolving. Might be included with another PR.